src/attr.rs

```diff
@@ -56,7 +58,7 @@ pub(crate) fn is_our_attr(attr: &Attribute) -> bool {
 /// attributes. If it's invalid, an error is emitted and `Err(())` is returned.
 /// You have to make sure that `attr` is one of our attrs with `is_our_attr`
 /// before calling this function!
-pub(crate) fn parse_our_attr(attr: &Attribute) -> Result<OurAttr, ()> {
+pub(crate) fn parse_our_attr(attr: &Attribute) -> Option<OurAttr> {
```

Comment:
I'd much prefer to keep `Result<T, ()>`. While it's obviously the same on machine level, it is conceptually different from `Option<T>`. `Err(())` clearly conveys that an error occurred, whereas `None` does not. Or did you have a special reason to change those?

This also affects a couple of other functions, which I won't comment on individually.